### ISSUE TITLE
Summernote sucks, replace with Trumbowyg

### ISSUE THREAD

**Issue by rmehta** [frappe/erpnext#13361]
https://alex-d.github.io/Trumbowyg/

**Comment by joolsr** [comment]
What's wrong with Summernote? Just interested in what you consider the faults are. Also it wasn't so long back that Summernote replaced the previous editor ..
But good to see improvements continuing ;-)

**Comment by dottenbr** [comment]
Summernote creates extremely messy HTML which is made worse by some specific code in Frappe which replaces <p> with <div>. I played around with Trumbowyg and it seems to create pretty clean code so yea, all for it!

**Comment by giuseppelamatrice** [comment]
+1

**Comment by netchampfaris** [comment]
I have tried implementing both Trumbowyg and CkEditor, here's a list of pros and cons. Additions welcome.

Trumbowyg:

Pros:
- Depends on jQuery

Cons
- Trumbowyg depends on `contenteditable` like any other WYSIWYG editor out there, even summernote which is the source of all problems.
- Not all features of summernote are included like emojis and mentions (will have to write plugins)
- Opinionated widgets and UI which can be overridden by introducing more styles.

CkEditor:

Pros
- Has an abstract MVC architecture over the DOM, so doesn't depend on `contenteditable`, hence it will render content the same on every browser.
- All of it's styling can be overridden by just changing CSS Variables

Cons
- Documentation/API difficult to understand, hence difficult to write plugins for it.
- Not all features of summernote are included like emojis and mentions (will have to write plugins).

**Comment by netchampfaris** [comment]
I think we should also have a markdown edit mode like GitHub.

**Comment by pratu16x7** [comment]
I have tried the Trumbowyg replacement, it's working is similar to Summernote being built upon `contentEditable` as @netchampfaris pointed out.

The fact that it certainly is less painful can be attributed to its simplicity, that is accounted for by the lack of features (beware, nearly every level 2 feature, even **fonts**, **tables**, and **emoji**) that will all need plugins.

/Users/prateekshasingh/Library/Group Containers/6N38VWS5BX.ru.keepcoder.Telegram/account-17204323211624500020/postbox/media/telegram-cloud-file-5-852840979-25859-5213047634931067938.jpg

Haven't tried CKEditor, but based on the pros and cons listed, it is a winner in being browser agnostic.

**Comment by rmehta** [comment]
@pratu16x7 @netchampfaris do we have both implemented?

**Comment by netchampfaris** [comment]
@rmehta Yes, but not merged yet. Also, I have closed my PR for now because webforms still treat the TextEditor separately, so I think we should wait for the new Webforms to get merged before implementing any of the two editors.

**Comment by rmehta** [comment]
The reason we need more communication 😉
